azureeyemodule/app/iot/iot_interface.cpp

```diff
@@ -200,6 +219,17 @@ static IOTHUBMESSAGE_DISPOSITION_RESULT receive_onvif_message(IOTHUB_MESSAGE_HAN
     auto deserialized_msg = parse_onvif_message(msg);
 
     // Set stuff based on results
+    if (deserialized_msg.find(std::string("snapshot")) != deserialized_msg.end())
+    {
+        const auto uri = deserialized_msg[std::string("snapshot")];
+	std::string cmd = "gst-launch-1.0 rtspsrc location=\"rtsp://localhost:" + uri + "\" is_live=true ! decodebin ! jpegenc snapshot=TRUE ! filesink location=/snapshot/snapshot.jpg";
+	if (system(cmd.c_str()) < 0)
```

Comment:
According to [the GStreamer documentation](https://gstreamer.freedesktop.org/documentation/tools/gst-launch.html?gi-language=c):

> Please note that gst-launch-1.0 is primarily a debugging tool. You should not build applications on top of it. For applications, use the gst_parse_launch() function of the GStreamer API as an easy way to construct pipelines from pipeline descriptions.

Since we already link against GStreamer, there doesn't seem to be a good reason not to just use gst_parse_launch instead of invoking gst-launch-1.0 via system().

Reply:
But it is not linked in the current iot_interface.cpp file. Does that mean I need to additionally include <gst/gst.h> in the file ?

Reply:
We link against the GStreamer library (that's done in the CMake file), but you *include* the header files you need wherever you need them.

Yeah, go ahead and include the gst.h header (and whatever other ones you need) in that .cpp file.